jenkins/opensearch/integ-test.jenkinsfile

```diff
@@ -0,0 +1,122 @@
+lib = library(identifier: "jenkins@20211118", retriever: legacySCM(scm))
+
+pipeline {
+    agent none
+    environment {
+        AWS_ROLE_ARN = "arn:aws:iam::${AWS_ACCOUNT_PUBLIC}:role/opensearch-test"
+        AWS_ROLE_SESSION_NAME = "jenkins-test-session"
+    }
+    tools {
+        jdk "JDK14"
+        maven "maven-3.8.2"
+    }
+    parameters {
+        string(
+            defaultValue: '',
+            name: 'TEST_MANIFEST',
+            description: 'Test manifest under the manifests folder, e.g. manifests/2.0.0/opensearch-2.0.0-test.yml.',
+            trim: true
+        )
+        string(
+            defaultValue: '',
+            name: 'BUILD_MANIFEST_URL',
+            description: 'The build manifest URL, e.g. https://ci.opensearch.org/ci/dbc/distribution-build-opensearch/1.2.2/98/linux/x64/dist/opensearch/manifest.yml.',
+            trim: true
+        )
+    }
+    stages {
+        stage('Extract build information from build manifest') {
+            steps {
+                node('Jenkins-Agent-al2-x64-m5xlarge') {
+                    script {
+                        getBuildManifestInfo()
+                    }
+                }
+            }
+        }
+        stage('Choose Agent Node') {
+            steps {
+                script {
+                    if ("${ARCHITECTURE}" == 'x64') {
+                        agentLabel = "Jenkins-Agent-al2-x64-m5xlarge"
+                    } else if ("${ARCHITECTURE}" == 'arm64') {
+                        agentLabel = "Jenkins-Agent-al2-arm64-m6gxlarge"
+                    } else {
+                        error("Unknown CPU architecture is provided")
+                    }
+                }
+            }
+        }
+        stage('integ-test') {
+            agent {
+                node {
+                    label "${agentLabel}"
+                }
+            }
+            steps {
+                script {
+                    runIntegTestScript(
+                        buildManifest: "${BUILD_MANIFEST}",
+                        testManifest: "${TEST_MANIFEST}"
+                    )
+                }
+            }
+            post {
+                always {
```

Comment:
Adding a try/catch means the build status won't be changed, which means as long as the tests run it will always show up as successful, which means the publish notifications steps below will always run the `success` branch regardless of tests succeeding/failing.

Are you suggesting parsing the results of running the `test.sh` script to determine success/failure? I feel like this can unnecessary work instead of leveraging jenkins failures to tell us whether the integ tests passed or failed. Lmk what you think.

Reply:
If you want the job to show failure, use `catchError` (its also mentioned in that SO link).

Maybe I am being very pessimistic as I expect the test results to fail often while we are first confronting our intermittent failing tests.  By not reporting failure unless there are workflow related failures - it would bring attention to them more quickly IMO.

I'll leave it up to you what route you would prefer.

Reply:
I would start by failing the job if integration tests fail. There's no hope to make them reliable if there aren't loud visible failures.

Reply:
@peternied I was hesitant to go with `catchError` since it was mentioned that it's deprecated. I prefer failing the job if the tests fail for now. Agreed that it may bring more failures first, but it will probably help motivate owners of the test cases to fix flaky / broken tests. If it gets too noisy we can easily throw in a try/catch in the future.